### ISSUE TITLE
"Convert to switch expression" refactoring for ternary statements

### ISSUE THREAD

**Issue by rabuckley** [dotnet/roslyn#75973]
**Brief description:**

Light up the "Convert to switch expression" refactoring for ternary statements.

This is a style preference.

**Code example that the analyzer should report:**

Currently, the code

```cs
static string? Case1(ReadOnlySpan<char> source)
{
    var i = source.IndexOf(':');

    var substring =  i < 0
        ? null
        : source[i..].ToString();

    return substring;
}
```

_Sequentially_ has the possible refactorings,

1. Replace conditional expression with statements
2. Convert to switch statement
3. Convert to switch expression

It would be nice if it would suggest all three in the first instance, rather than requiring one or two transformations first. I.e., suggest the refactoring from ternary to,

```cs
static string? Case1(ReadOnlySpan<char> source)
{
    var i = source.IndexOf(':');

    var substring = i switch
    {
        < 0 => null,
        _ => source[i..].ToString(),
    };

    return substring;
}
```

**Comment by CyrusNajmabadi** [comment]
Not sure how i feel about this.  Conditional is much terser and more idiomatic here for me.  If anything, i'd support moving a two-branch switch expr to a conditional expr.
